### PR TITLE
feat: 1405 - Create offer

### PR DESCRIPTION
**Pull request by momentiris**
- Adds process code to create-offer (excluding notifying offered applicant)
- Separate internal parking space processes into files
- Updates `ProcessResult` type to a union of `ProcessSuccess<T> | ProcessError<E>` so consumer can tell exactly what went wrong. Only applied fully to create-offer so far but other processes were affected as well so there is minor changes to those. I think it's better to include typed error reasons in the processes so the consumer can decide for themselves how to handle the different scenarios. In the future i'm thinking maybe processes shouldn't return httpStatus neither. Why  should the process decide what the consumer should do?

Fixes https://dev.azure.com/mimeronline/ONECore/_boards/board/t/Uthyrning/Issues/?workitem=1405

### CODE REVIEW

File: src/adapters/leasing-adapter.ts [hunk 1]
```diff
@@ -303,9 +306,10 @@ const getApplicantByContactCodeAndRentalObjectCode = async (
   }
 }
 
+// TODO: This function does not actually get the listing
```
Comment (mittistormen) [inline src/adapters/leasing-adapter.ts:309]: Gör den inte? :)

File: src/common/types.ts [hunk 2]
```diff
@@ -1,11 +1,20 @@
 export enum ProcessStatus {
   successful,
   failed,
-  inProgress,
 }
 
-export interface ProcessResult {
+export interface ProcessSuccess<T = any> {
```
Comment (mittistormen) [inline src/common/types.ts:6]: 👏🏻

**Review by mittistormen — APPROVED** [review]

File: src/adapters/leasing-adapter.ts (hunk 1 above)
Comment (momentiris) [reply src/adapters/leasing-adapter.ts:309]: Nej den hämtar `Array<DetailedApplicant>` baserat på listing id. Därfår behöver jag i processen först hämta `getListingByListingId` och sen `getListingByIdWithDetailedApplicants` för att hämta prioriterade listan av applicants. Baserat på namnen borde det räckt med det senare anropet där men då det inte hämtar själva listingen så får vi göra båda två.

Comment (mittistormen) [reply src/adapters/leasing-adapter.ts:309]: Aha där ser man

**Pull request merged**